editor/src/components/inspector/sections/component-section/property-control-controls.tsx

```diff
@@ -381,3 +394,50 @@ export const ControlForStringProp = betterReactMemo(
     )
   },
 )
+
+export const ControlForVectorProp = betterReactMemo(
+  'ControlForVectorProp',
+  (props: ControlForPropProps<Vector2ControlDescription | Vector3ControlDescription>) => {
+    const { propPath, propMetadata, controlDescription } = props
+
+    const vectorValue =
+      (propMetadata.propertyStatus.set ? propMetadata.value : controlDescription.defaultValue) ?? []
+
+    const keys = controlDescription.type === 'vector3' ? ['x', 'y', 'z'] : ['x', 'y']
```

Comment:
@enidemi how confident are you that vector3 (in context of react-three-fiber) can only be "x, y, z"?

Reply:
This is what I found in the three js docs https://threejs.org/docs/#api/en/math/Vector3

Reply:
nice! can you add this as a comment to the Vector3 interface in property-controls?

Reply:
For vector3 I've seen some samples that use their third definition ("Any arbitrary ordered triplet of numbers"), but the vast majority are xyz. vector2 seems to be the more open-ended one but even there it's prob 80/20.

in conclusion: as long as we don't _force_ these for labels, I'd be ok _defaulting_ to them for labels. Does that work with our property controls?

Reply:
That would require changing the definition to include labels and provide defaults to them. Looking at that definition (and the one of `Vector2`), they even use x,y,z as the constructor param names.